### PR TITLE
Fixes some issues with medical gloves

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
I was informed the medical gloves have some issues. Namely, being usable on all intents, and the scanner gloves not being able to show patients their scan via right click. This PR fixes those issues, and I tested it. 

## Why It's Good For The Game
QoL and more standard / expected behavior. 
Also you won't scan people when you punch them anymore. 

## Changelog

:cl:
fix: fixed medical gloves triggering on all intents
fix: fixed medical scanner gloves not having the right click behavior when worn
/:cl:

### CODE REVIEW

File: code/game/objects/items/scanners.dm [hunk 1]
```diff
@@ -297,17 +297,42 @@ REAGENT SCANNER
...
+	. = ..()
+	if(user.a_intent != INTENT_HELP)
+		return
+	analyze_vitals(M, user)
+
+/obj/item/healthanalyzer/gloves/attack_alternate(mob/living/carbon/M, mob/living/user)
+	. = ..()
+	if(user.a_intent != INTENT_HELP)
+		return
+	analyze_vitals(M, user, TRUE)
```
Comment (QualityVan) [inline code/game/objects/items/scanners.dm:334]: Is this for slapping people with the gloves? That seems like a weird thing to add, but I guess it works?

File: code/game/objects/items/scanners.dm [hunk 2]
```diff
@@ -297,17 +297,42 @@ REAGENT SCANNER
...
 //when you are wearing these gloves, this will call the normal attack code to health scan the target
 /obj/item/healthanalyzer/gloves/proc/on_unarmed_attack(mob/living/carbon/human/user, mob/living/carbon/human/target)
-	if(istype(user) && istype(target))
-		attack(target,user)
+	if(user.a_intent != INTENT_HELP)
+		return
+	else if(istype(user) && istype(target))
+		analyze_vitals(target, user)
+
+/obj/item/healthanalyzer/gloves/proc/on_unarmed_attack_alternate(mob/living/carbon/human/user, mob/living/carbon/human/target)
```
Comment (QualityVan) [inline code/game/objects/items/scanners.dm:318]: Autodocs here and in defib.

**Review by QualityVan — CHANGES_REQUESTED** [review]
These signals seem like they should interrupt the normal attack_hand calls, wondering if I'm missing a particular reason they don't.

File: code/game/objects/items/scanners.dm (hunk 1 above)
Comment (lbnesquik) [reply code/game/objects/items/scanners.dm:334]: I figured that could be desired, so...

File: code/game/objects/items/scanners.dm (hunk 2 above)
Comment (lbnesquik) [reply code/game/objects/items/scanners.dm:319]: done

File: code/game/objects/items/scanners.dm [hunk 3]
```diff
@@ -297,17 +297,43 @@ REAGENT SCANNER
...
+	UnregisterSignal(user, COMSIG_HUMAN_MELEE_UNARMED_ATTACK_ALTERNATE) //Unregisters in the case of getting delimbed
+
 
 //when you are wearing these gloves, this will call the normal attack code to health scan the target
 /obj/item/healthanalyzer/gloves/proc/on_unarmed_attack(mob/living/carbon/human/user, mob/living/carbon/human/target)
-	if(istype(user) && istype(target))
-		attack(target,user)
+	if(user.a_intent != INTENT_HELP)
+		return
+	else if(istype(user) && istype(target))
```
Comment (ivanmixo) [inline code/game/objects/items/scanners.dm:315]:
```suggestion
	if(istype(user) && istype(target))
```

File: code/game/objects/items/scanners.dm [hunk 4]
```diff
@@ -297,17 +297,43 @@ REAGENT SCANNER
...
+		analyze_vitals(target, user)
+
+///Used for right click and showing the patient their scan
+/obj/item/healthanalyzer/gloves/proc/on_unarmed_attack_alternate(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	if(user.a_intent != INTENT_HELP)
+		return
+	else if(istype(user) && istype(target))
```
Comment (ivanmixo) [inline code/game/objects/items/scanners.dm:322]:
```suggestion
	if(istype(user) && istype(target))
```

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Nitpicky but for future references `else` is completely worthless if the above `if` statement has a `return` at the end

**Comment by lbnesquik** [comment]
> Nitpicky but for future references `else` is completely worthless if the above `if` statement has a `return` at the end

I forgor

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**